packages/webview_flutter/webview_flutter_platform_interface/lib/src/v4/navigation_callback_handler_delegate.dart

```diff
@@ -0,0 +1,83 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+
+import 'webview_platform.dart';
+
+/// Interface for callbacks made by [NavigationCallbackHandlerDelegate].
+///
+/// The webview plugin implements this class, and passes an instance to the
+/// [NavigationCallbackHandlerDelegate].
+/// [NavigationCallbackHandlerDelegate] is notifying this handler on events that
+/// happened on the platform's webview.
+abstract class NavigationCallbackHandlerDelegate extends PlatformInterface {
+  /// Creates a new [NavigationCallbacksHandlerDelegate]
+  factory NavigationCallbackHandlerDelegate() {
+    final NavigationCallbackHandlerDelegate callbackHandlerDelegate =
+        WebViewPlatform.instance!.createNavigationCallbackHandlerDelegate();
+    PlatformInterface.verify(callbackHandlerDelegate, _token);
+    return callbackHandlerDelegate;
+  }
+
+  /// Used by the platform implementation to create a new
+  /// [NavigationCallbackHandlerDelegate].
+  ///
+  /// Should only be used by platform implementations because they can't extend
+  /// a class that only contains a factory constructor.
+  @protected
+  NavigationCallbackHandlerDelegate.implementation() : super(token: _token);
+
+  static final Object _token = Object();
+
+  /// Sets the callback method that is invoked by the
+  /// [WebViewPlatformControllerDelegate] when a navigation request is pending.
```

Comment:
This and the documentation for the methods below should have a one line summary. 

I think something like

```dart
/// Invoked when a navigation request is pending.
///
/// See [WebViewControllerDelegate.setNavigationCallbackHandler].
```


https://dart.dev/guides/language/effective-dart/documentation#do-separate-the-first-sentence-of-a-doc-comment-into-its-own-paragraph